forte_wrapper/allennlp/allennlp_processors.py

```diff
@@ -139,20 +139,30 @@ def _process(self, input_pack: DataPack):
         self._process_existing_entries(input_pack)
 
         batch_size: int = self.configs['infer_batch_size']
-        batches: Iterator[Iterable[Sentence]]
+        batches: Iterator[Iterator[Sentence]]
+        # Need a copy of the one-pass iterators to support a second loop on
+        # them. All other ways around it like using `itertools.tee` and `list`
+        # would require extra storage conflicting with the idea of using
+        # iterators in the first place. `more_itertools.ichunked` uses
+        # `itertools.tee` under the hood but our usage (reading iterators
+        # in order) does not cause memory issues.
+        batches_copy: Iterator[Iterator[Sentence]]
         if batch_size <= 0:
             batches = iter([input_pack.get(Sentence)])
+            batches_copy = iter([input_pack.get(Sentence)])
         else:
-            batches = more_itertools.chunked(
+            batches = more_itertools.ichunked(
```

Comment:
This copy fix looks a bit complex, how about this?

```suggestion
            batches = more_itertools.chunked(list(input_pack.get(Sentence)), batch_size
```

Reply:
nevermind